trieste/acquisition/rule.py

```diff
@@ -1103,6 +1115,361 @@ def state_func(
         return state_func
 
 
+class UpdateableSearchSpace(SearchSpace):
+    """A search space that can be updated."""
+
+    @abstractmethod
+    def reinitialize(
+        self,
+        models: Optional[Mapping[Tag, ProbabilisticModelType]] = None,
+        datasets: Optional[Mapping[Tag, Dataset]] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Reinitialize the search space using the given models and datasets."""
+        ...
+
+    @abstractmethod
+    def update(
+        self,
+        models: Optional[Mapping[Tag, ProbabilisticModelType]] = None,
+        datasets: Optional[Mapping[Tag, Dataset]] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Update the search space using the given models and datasets."""
+        ...
+
+    def get_single_model_and_dataset(
```

Comment:
Should this be a standalone function? Perhaps even a simple generic one?

```python
def get_tag(mapping: Optional[Mapping[Tag, T]], tag: Tag = OBJECTIVE) -> Optional[T]:
    if mapping is None:
        return None
    elif tag in mapping.keys():
        return mapping[tag]
    else:
        raise ValueError(f"tag {tag} not found in mapping")
```

which you can then use as `get_tag(models), get_tag(datasets)` (or elsehow as necessary).